server/hotrod/src/main/java/org/infinispan/server/hotrod/CacheDecodeContext.java

```diff
@@ -149,11 +149,14 @@ Response get() {
    Response getKeyMetadata() {
       CacheEntry<byte[], byte[]> ce = cache.getCacheEntry(key);
       if (ce != null) {
-         NumericVersion entryVersion = (NumericVersion) ce.getMetadata().version();
+         long version = 0;
+         EntryVersion entryVersion = ce.getMetadata().version();
+         if (entryVersion != null && entryVersion instanceof NumericVersion) {
```

Comment:
>  I personally have never seen SimpleClusteredVersion in use or in handling in code.

Apparently the server sets this version scheme under the hood on caches that 
`mode.isSynchronous() && lockingMode == LockingMode.OPTIMISTIC && isolationLevel == IsolationLevel.REPEATABLE_READ`.

> I was saying to just use the version from the SimpleClusteredVersion as we use the NumericVersion

yes, that'd be the only way to handle the SimpleClusteredVersion in the current protocol version, but as I was saying, it would loose precision or whatever since the `topologyId` bit of the `SimpleClusteredVersion` is significant to perform comparisons.